qa/integration-tests/src/test/java/io/zeebe/broker/it/workflow/CancelWorkflowInstanceTest.java

```diff
@@ -111,18 +112,21 @@ public void shouldFailToCompleteJobAfterCancel() {
 
     waitUntil(() -> jobEvents.size() > 0);
 
-    clientRule.getWorkflowClient().newCancelInstanceCommand(workflowInstance).send().join();
+    clientRule
+        .getWorkflowClient()
+        .newCancelInstanceCommand(workflowInstance.getWorkflowInstanceKey())
+        .send()
+        .join();
 
     // when
     assertThatThrownBy(
             () -> {
-              clientRule.getJobClient().newCompleteCommand(jobEvents.get(0)).send().join();
+              clientRule.getJobClient().newCompleteCommand(jobEvents.get(0).getKey()).send().join();
             })
-        .isInstanceOf(ClientCommandRejectedException.class);
+        .isInstanceOf(ClientException.class);
 
     // then
-    waitUntil(() -> eventRecorder.hasJobEvent(JobState.CANCELED));
-    waitUntil(
-        () -> eventRecorder.hasElementInState("process", WorkflowInstanceState.ELEMENT_TERMINATED));
+    assertThat(RecordingExporter.jobRecords(JobIntent.CANCELED).exists()).isTrue();
```

Comment:
Replace by a assertion helper method `assertIncidentCanceled`

Reply:
Why incident canceled? Besides that incident has no cancel